app/test/testMetadata.js

```diff
@@ -0,0 +1,192 @@
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+const fetchMock = require('fetch-mock');
+const {app} = require('../service');
+
+chai.use(chaiHttp);
+chai.should();
+
+const exampleMetadata = {
+  description: 'An example description',
+  favicon_url: 'http://www.example.com/rich-icon.png',
+  icon_url: 'http://www.example.com/rich-icon.png',
+  image_url: 'http://www.example.com/preview.png',
+  images: [
+    {
+      'url': 'http://www.example.com/preview.png',
+      'entropy': 1,
+      'height': 500,
+      'width': 500,
+    }
+  ],
+  original_url: 'http://www.example.com/good',
+  provider_url: 'http://www.example.com/good',
+  title: 'An Example Page',
+  url: 'http://www.example.com/good'
+};
+
+const goodExampleUrl = 'http://www.example.com/good';
+const badExampleUrl = 'http://www.example.com/bad';
+
+
+describe('Metadata API Tests', function() {
+  before(function() {
+    fetchMock.mock(
+      goodExampleUrl,
+      `
+        <html>
+          <head>
+            <title>${exampleMetadata.title}</title>
+            <meta name="description" content="${exampleMetadata.description}" />
+            <meta name="thumbnail" content="${exampleMetadata.images[0].url}" />
+            <link rel="icon" href="http://www.example.com/rich-icon.png" />
+          </head>
+          <body></body>
+        </html>
+      `
+    );
+  });
+
+  after(function() {
+    fetchMock.restore();
+  });
+
+  it('should raise 400 if missing JSON body', (done) => {
+    chai.request(app)
+      .post('/')
+      .end((err, res) => {
+        res.should.have.status(400);
+        done();
+      });
+  });
+
+  it('should raise 400 if missing urls parameter', (done) => {
+    chai.request(app)
+      .post('/')
+      .send(JSON.stringify({}))
+      .end((err, res) => {
+        res.should.have.status(400);
+        done();
+      });
+  });
```

Comment:
Same comment as below. Does this fail w/ status 400 because of missing URLs parameter, or missing content header?

```sh
$ curl -XPOST -i -H "content-type: application/json" -d '{}' http://localhost:7001 | JSON

HTTP/1.1 400 Bad Request
X-Powered-By: Express
Content-Type: application/json; charset=utf-8
Content-Length: 57
ETag: W/"39-DATCOl+kpDISCiW+0MhzZQ"
Date: Tue, 09 Aug 2016 23:26:02 GMT
Connection: keep-alive

{
  "error": "Unable to locate 'urls' in body: {}",
  "urls": []
}
```

Instead of validating just the status code, not sure if we want to also try verifying the `error` response string from the server.